### DISCUSSION TITLE
Animations lag with Tailwind on IOS

### DISCUSSION THREAD

**Discussion by FabioSimoDev** [tailwindlabs/tailwindcss#13313] [Help] [answered]
I created a Vite + React project. Inside, I created a 'typewriter' animation, which uses JavaScript to animate the text. Initially, I created this animation using intervals and timeouts, and it worked correctly on desktop but was choppy on mobile, causing delays and more letters appearing together. Assuming this was due to the difference in frame rate between desktop and mobile, I modified the code using requestAnimationFrame. This improved the visual animation by removing the lag, but the slowness remained. The animation was supposed to take about 3 seconds to complete, but it took at least 5 or 6. In another desperate attempt, I used a mix of Promise and Timeout, but the initial problem returned. At this point I couldn't find any other solutions, so I created an empty project and gradually added all the components to see how the animation reacted. The animation was smooth with all the components and logic added, only Tailwind was left to add. 
_**As soon as I installed it and restarted the project**_, I noticed the same problem returned: the animation was slow and choppy. My style is very simple; I don't have any other particular animations or dynamic things. I've been trying for days; I even tried using `experimental: { optimizeUniversalDefaults: true }` but the problem remains. Does anyone know what could be causing this?

**Without Tailwind:**
![trim](https://github.com/tailwindlabs/tailwindcss/assets/143503265/fd5e5158-407f-4943-ad17-a643b7e32f35)

**With Tailwind:**
![trim(1)](https://github.com/tailwindlabs/tailwindcss/assets/143503265/e5ccc27b-eb16-487a-9156-cf5c509d90ca)

tested on IOS, Safari, Chrome and Opera

**Comment by FabioSimoDev** [comment]
Update:
After conducting further tests, I noticed that the issue only occurs on **iOS devices**. I tested it on an old Android phone I had at home, and the animation is smooth. If it's smooth on that old phone, it should be smooth everywhere. So it seems to be a problem caused by Tailwind on iOS devices. Any solutions? Is it a known issue?

**Reply by ludofischer** [reply]
Does the same problem occur if you style the text using the same CSS properties, but without using Tailwind?

**Reply by FabioSimoDev** [reply, marked as answer]
Thanks for the response. 
As I mentioned, I had run some tests and thought the answer to this question was yes, but actually it's no. It's a strange issue. In the background I have two divs with different colors to which a blur is applied. In the new test project I tried applying the blur to only one of these two div to understand if the problem stemmed from that. For some reason, I discovered that the issue arises from the div on which the blur is applied; everything around it becomes slow and laggy. From what I've seen online, it's a known issue on Safari that, for some reason, hasn't been resolved yet. I managed to 'mask' the problem by using the `transform: translate3d(0, 0, 0)` property on the elements with blur to force the browser to use the GPU instead of the CPU. 
So, in the end, the problem wasn't related to Tailwind. Apologies for the wasted time
